### DISCUSSION TITLE
Version did not increment as expected after first release to main

### DISCUSSION THREAD

**Discussion by LukeHersman** [semantic-release/semantic-release#3471] [Release workflow] [answered]
Hi all!

We have a configuration with a main branch and a develop branch. Develop branch is configured as a pre-release branch. Before our first release, everything was working as expected. The versions were all something like v1.0.0-beta.1, v1.0.0-beta.2, v1.0.0-beta.3, ect. We just had our first merge to main, which produced v1.0.0 as expected. 

However, on our next `feat` merged into develop produced v1.0.0-beta.4 instead of the expected v1.1.0-beta.1. Can you see any issue with our configuration or some other reason this would have occurred? 

```
"release": {
    "branches": [
        {
            "name": "main"
        },
        {
            "name": "develop",
            "channel": "beta",
            "prerelease": "beta"
        }
    ],
    "plugins": [
        "semantic-release-export-data",
        "@semantic-release/commit-analyzer",
        "@semantic-release/release-notes-generator",
        [
            "@semantic-release/changelog",
            {
                "changelogFile": "CHANGELOG.md"
            }
        ],
        [
            "@semantic-release/npm",
            {
                "npmPublish": false
            }
        ],
        [
            "@semantic-release/git",
            {
                "assets": [
                    "CHANGELOG.md",
                    "package.json"
                ],
                "message": "chore(release): set `package.json` to ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}"
            }
        ],
        [
            "@semantic-release/github",
            {
                "assets": [
                    {
                        "path": "build_output/*.tar.xz",
                        "label": "Linux distribution"
                    },
                    {
                        "path": "build_output/*.zip",
                        "label": "Windows distribution"
                    }
                ]
            }
        ]
    ]
},
```

**Comment by travi** [comment, marked as answer]
your question directly highlights one of the reasons we do not officially support gitflow. semantic-release determines the next version based on the versions in the history of the branch it is currently acting on (based on historical tags and git notes). 

it sounds like you merged the develop branch into your default branch and had a successful release there. then you made additional commits to the develop branch expecting it to know about the release that happened in your default branch. without merging the default branch back into the develop branch, semantic-release does not know this information. this is by design so that versions maintain proper lineage. if you want to simulate gitflow with our pre-release workflows, you need to, at minimum, merge the stable release from you default branch back into your develop branch.

the detail worth understanding here is that this _is not_ the intended use of the pre-release workflow. it is instead intended for temporary pre-releases to get feedback from consumers about api design details. it is not a testing workflow as gitflow enables. semantic-release encourages continuous integration and continuous deployment and expects to operate in a context where releases are not necessary that behavior is working as intended.

**Reply by LukeHersman** [reply]
That makes sense, thank you for the reply!
